libs/providers/vmware.py

```
@@ -333,6 +333,36 @@ def stop_vm(self, vm):
+        cluster = host.parent
+        if not isinstance(cluster, vim.ClusterComputeResource):
+            raise VmCloneError(f"VM '{vm.name}' host is not in a ClusterComputeResource; cannot disable DRS per-VM")
+        drs_vm_config = vim.cluster.DrsVmConfigInfo(key=vm, enabled=False)
+        vm_config_spec = vim.cluster.DrsVmConfigSpec(
+            info=drs_vm_config,
+            operation=vim.option.ArrayUpdateSpec.Operation.add,
+        )
+        cluster_spec = vim.cluster.ConfigSpecEx(drsVmConfigSpec=[vm_config_spec])
+        task = cluster.ReconfigureComputeResource_Task(cluster_spec, modify=True)
+        self.wait_task(task, f"Disable DRS for VM '{vm.name}'")
+        LOGGER.info(f"DRS disabled for VM '{vm.name}' on cluster '{cluster.name}'")
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)